src/main/java/teammates/ui/controller/InstructorCourseStudentDetailsPageData.java

```diff
@@ -1,19 +1,43 @@
 package teammates.ui.controller;
 
 import teammates.common.datatransfer.AccountAttributes;
-import teammates.common.datatransfer.InstructorAttributes;
+import teammates.common.datatransfer.StudentAttributes;
 import teammates.common.datatransfer.StudentProfileAttributes;
+import teammates.common.util.Const;
+import teammates.ui.template.StudentInfoTable;
+import teammates.ui.template.StudentProfile;
 
 public class InstructorCourseStudentDetailsPageData extends PageData {
     
-    public InstructorAttributes currentInstructor;
-    public StudentProfileAttributes studentProfile;
-    public String regKey;
-    public boolean hasSection;
-    public String commentRecipient;
+    private StudentProfile studentProfile;
+    private StudentInfoTable studentInfoTable;
+    private String commentRecipient;
 
-    public InstructorCourseStudentDetailsPageData(AccountAttributes account) {
+    public InstructorCourseStudentDetailsPageData(AccountAttributes account, StudentAttributes student,
+            StudentProfileAttributes studentProfile, boolean isAbleToAddComment, boolean hasSection,
+            String commentRecipient) {
         super(account);
+        if (studentProfile != null) {
+            String pictureUrl = studentProfile.pictureKey == null || studentProfile.pictureKey.isEmpty()
+                              ? Const.SystemParams.DEFAULT_PROFILE_PICTURE_PATH
+                              : Const.ActionURIs.STUDENT_PROFILE_PICTURE + "?"
+                                + Const.ParamsNames.BLOB_KEY + "=" + studentProfile.pictureKey + "&"
+                                + Const.ParamsNames.USER_ID + "=" + account.googleId;
+            this.studentProfile = new StudentProfile(student.name, studentProfile, pictureUrl);
```

Comment:
I know, the general one isn't coupled with StudentProfileAttributes at the
moment, I believe? That's what I meant.
On 29 Jun 2015 12:30, "Wilson Kurniawan" <notifications@github.com> wrote:

> In
> src/main/java/teammates/ui/controller/InstructorCourseStudentDetailsPageData.java
> <https://github.com/TEAMMATES/repo/pull/3828#discussion_r33433377>:
>
> >          super(account);
> > +        if (studentProfile != null) {
> > +            String pictureUrl = studentProfile.pictureKey == null || studentProfile.pictureKey.isEmpty()
> > +                              ? Const.SystemParams.DEFAULT_PROFILE_PICTURE_PATH
> > +                              : Const.ActionURIs.STUDENT_PROFILE_PICTURE + "?"
> > +                                + Const.ParamsNames.BLOB_KEY + "=" + studentProfile.pictureKey + "&"
> > +                                + Const.ParamsNames.USER_ID + "=" + account.googleId;
> > +            this.studentProfile = new StudentProfile(student.name, studentProfile, pictureUrl);
>
> No I mean to the general PageData class. Then anything inheriting PageData
> can reuse that as well. You know, like those link generator methods.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/TEAMMATES/repo/pull/3828/files#r33433377>.
>